File: JavaScript/1768. Merge Strings Alternately.js
//  1768. Merge Strings Alternately
//  Difficulty: Easy
// Submission Link: https://leetcode.com/problems/merge-strings-alternately/submissions/1016823533/

var mergeAlternately = function(word1, word2) {
    let len1 = word1.length;
    let len2 = word2.length;
    
    let merged = '';
    let i = 0, j = 0;
    
    while (i < len1 && j < len2) {
        merged += word1[i++] + word2[j++];
    }
    
    while (i < len1) {
        merged += word1[i++];
    }
    
    while (j < len2) {
        merged += word2[j++];
    }
    
    return merged;
};

File: Python3/1768. Merge Strings Alternately.py
""" 1768. Merge Strings Alternately
Difficulty: Easy
Submission Link: https://leetcode.com/problems/merge-strings-alternately/submissions/1375628956/ """

class Solution:
    def mergeAlternately(self, word1: str, word2: str) -> str:
        merged = []

        for a, b in zip(word1, word2):
            merged.append(a + b)
        
        merged.append(word1[len(word2):])
        merged.append(word2[len(word1):])

        return "".join(merged)